### ISSUE TITLE
Not update schema.rb when run with: rake db:drop db:create db:migrate

### ISSUE THREAD

**Issue by zw963** [rails/rails#18464]
## My Case

I add a hstore field to my products table with  
```ruby
class AddAdditionalFieldsToProducts < ActiveRecord::Migration
  def change
    add_column :products, :additional_fields, :hstore
    add_index :products, :additional_fields, using: 'gin'
  end
end
```
It will add a new fields to my products table.

schema.rb like this:

```ruby
  create_table "products", force: true do |t|
    t.hstore   "additional_fields",                          default: ""
  end
```

And then, I found I need a default value, so, I change my migrate to this.
(Although I know I should remove add readd a new migrate ... )

```ruby
class AddAdditionalFieldsToProducts < ActiveRecord::Migration
  def change
    add_column :products, :additional_fields, :hstore, default: {}
    add_index :products, :additional_fields, using: 'gin'
  end
end
```

Then, I run rake db:drop db:create db:migrate, it worked!
I have seen my products `additional_fields` default value is changed
to {} in rails console.

## Problem
But it not update my schema.rb with new {} default.

## Work Around
If i add one new migrate then db:migrate, it will update default value.
But db migrate history is not synchronize with my git commit.

**Comment by zw963** [comment]
Perhaps this is not bugs?  
before i add one new migration, migration history timestamp is not changed.

**Comment by zw963** [comment]
Hi, all, I reopen this issue for some reason:

## After Work Around

After I add a new migrate, then run rake db:migrate, 
it will update default to {}, it is Cool.

## New Problem

When I `rake db:drop db:create db:migrate`, it will restore additional_fields default
value to "", if I add a new migrate, will restore to {} again. But if I drop again ... 

How to fix this? I had add more migrate files after the modified migrate.
I don't know way `rake db:drop db:create db:migrate` is work not same as `db:migrate`
I don't want my db/schema.rb be changed unexpected.

I just needs a way to correct my mistake make by me previously. Thank you!

**Comment by zw963** [comment]
I don't want remove additional_fields and recreate this fields, Because some this field exist
useful data.

and seem like ``change_column_default :products, :additional_fields, {}`` not worked for me.

**Comment by zw963** [comment]
I don't know why in any case

`additional_fields`(hstore type field) default value will become to {} after I add a new migrate file.
But if i `rae db:drop db:create db:migrate`, default value wil become ""  again.
and, if add new migrate, default value will {} again 
......

I am so crazy for this...

**Comment by zw963** [comment]
I have tried to remove this hstore fields with following:

```ruby
class ChangeDefaultValueForAdditionalFieldsInProducts < ActiveRecord::Migration
  def change
    remove_column :products, :additional_fields
    add_column :products, :additional_fields, :hstore, default: {}
    add_index :products, :additional_fields, using: 'gin'
  end
end
```

after this migrate, in schema.rb, default value change to {}.

But when I  `rake db:drop db:create db:migrate`, default value change to ``""`` again.

FAINT ...

**Comment by senny** [comment]
@zw963 can you post the output when running `rake db:drop db:create db:migrate`? Also this pattern is contained within `bin/rake db:migrate:reset`.

**Comment by zw963** [comment]
Hi, I update my Project to Rails 4.2(ruby 2.2, pg 9.4) in my new year holiday, this problem is have resolved in ActiveRecord 4.2. (it always generate correct default value ``{}`` in schema.rb).

I try to restore to Rails 4.1 (with pg 9.4 and ruby 2.2), it reappear again, so I thought this bug is fixed in ActiveRecord 4.2,  Cool !

so, is still need to paste my rake db:reset output? it is a very long log...

If you still need,  I will post after holiday.

**Comment by senny** [comment]
@zw963 going to close the issue then. Thank you for reporting.
